ex2/ex2.js

```diff
@@ -1,32 +1,43 @@
 /*
  * Write a function that accepts a string and a number (n) and returns the index of the n-th comma in that string.
- * If there's no such comma, the function should return -1. If the n argument is invalid, the function should throw an Error.
+ * If there's no such comma, the function should return -1. If the n argument is incommaIndexid, the function should throw an Error.
  * For example:
  *     fn("hello,world,this,is,a,great,day", 3) === 16
  *     fn("hello,world,this,is,a,great,day", 4) === 19
  *     fn(",,,", 1) === 0
  *     fn(",,,", 4) === -1
  *     fn("hel,lowo,rld", 0) - Throws
  */
-function getTheCommaIndexAt(str, val) {
-	var index = -1, returnVal = 'Error';
-	for (var i = 0; i < val; i++) {
+function getTheCommaIndexAt(str, commaIndex) {
+	var index = -1,
+	for (var i = 0; i < commaIndex; i++) {
 		index = str.indexOf(',',index+1);
-		if (i === val-1 && index > -1) {
-			return index;
+		if (i === commaIndex-1 && index > -1) {
+			break;
 		} else if (index === -1) {
-			return -1;
+			break;
 		}
 	}
-	return returnVal;
+	return index;
 }
 
-function testFunction(str,val,msg) {
-	console.log(msg, getTheCommaIndexAt(str, val));
+function testFunction(str,commaIndex,msg) {
+
+	try{ 
+		if (commaIndex == 0) {
+			throw new RangeError('Throws');
```

Comment:
This should be in the main function, not the test function. You should test for this behavior here instead.